hours/importer/kirjastot.py

```diff
@@ -222,84 +292,363 @@ def get_openings(self, data: list) -> list:
+            if kirkanta_period["validUntil"]:
+                valid_until = parse(kirkanta_period["validUntil"]).date()
+
+            if valid_from is not None and valid_until is not None:
+                time_delta = valid_until - valid_from
+                if time_delta.days < 7:
+                    override_periods.append(kirkanta_period["id"])
+                    continue
+
+            period_schedules = [
+                i for i in schedules if i.get("period") == kirkanta_period["id"]
+            ]
+
+            if all([d["closed"] for d in period_schedules]):
+                override_periods.append(kirkanta_period["id"])
+
+            if kirkanta_period["isException"]:
+                override_periods.append(kirkanta_period["id"])
+
+        opening_hours = library.get_daily_opening_hours(start_date, end_date)
+
+        for schedule in schedules:
+            time_elements = []
+
+            if schedule.get("closed") is True:
+                time_elements.append(
+                    TimeElement(
+                        start_time=None,
+                        end_time=None,
+                        resource_state=State.CLOSED,
+                        override=True
+                        if schedule.get("period") in override_periods
+                        else False,
+                        full_day=True,
+                    )
+                )
+            else:
+                for schedule_time in schedule.get("times"):
+                    try:
+                        start_time = datetime.strptime(
+                            schedule_time.get("from"), "%H:%M"
+                        ).time()
+                    except ValueError:
+                        start_time = None
+                    try:
+                        end_time = datetime.strptime(
+                            schedule_time.get("to"), "%H:%M"
+                        ).time()
+                    except ValueError:
+                        end_time = None
+
+                    time_elements.append(
+                        TimeElement(
+                            start_time=start_time,
+                            end_time=end_time,
+                            resource_state=KIRKANTA_STATUS_MAP[schedule_time["status"]],
+                            override=True
+                            if schedule.get("period") in override_periods
+                            else False,
+                            full_day=False,
+                        )
+                    )
+
+            schedule_date = schedule.get("date")
+            if not isinstance(schedule_date, date):
+                schedule_date = parse(schedule.get("date")).date()
+
+            time_elements = combine_element_time_spans(time_elements)
+
+            time_elements.sort(key=self._get_times_for_sort)
+            opening_hours[schedule_date].sort(key=self._get_times_for_sort)
+
+            assert time_elements == opening_hours[schedule_date]
```

Comment:
Probably we need to use something else than `assert` here, since production does not necessarily run with assertions turned on, and we probably always want to raise an exception in production if this happens.